src/contexts/Cart.tsx

```diff
@@ -20,16 +23,63 @@ export const startingCart = {
       quantity: 1,
     },
   ],
-
-  setCart: (): void => undefined,
+  removeItem: (): void => undefined,
+  changeQuantity: (): void => undefined,
+  addItem: (): void => undefined,
 };
 
+export const removeItemFactory =
+  (setCart: React.Dispatch<React.SetStateAction<Cart>>) =>
+  (id: string): void => {
+    setCart((currCart) => currCart.filter((cartItem) => id !== cartItem.id));
+  };
+
+export const changeQuantityFactory =
+  (setCart: React.Dispatch<React.SetStateAction<Cart>>) =>
+  (id: string, amount: number): void => {
+    setCart((currCart) => {
+      return currCart.map((currItem) => {
+        if (currItem.id === id) {
+          const newQty =
+            currItem.quantity + amount > 0 ? currItem.quantity + amount : 0;
```

Comment:
Now that I'm thinking, when we reach zero, we need to filter that element out of the list

Reply:
So we don't see the item quantity hit 0,  it removes the item from cart instead?

Reply:
It's more a design choice in this case than correct implementation, so I'll leave this one up to you! Whatever you decide to do, just make a comment on the ticket so the testers know what behaviour to expect :)

Reply:
ok thanks Erica, will do!